src/use-auth.js

```diff
@@ -0,0 +1,86 @@
+import React, {
+  useState,
+  useContext,
+  createContext,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useQuery, useMutation, gql } from '@apollo/client';
+
+const REGISTER = gql`
+  mutation RegisterMutation(
+    $username: String!
+    $firstName: String!
+    $lastName: String!
+    $password: String!
+  ) {
+    createUser(
+      username: $username
+      firstName: $firstName
+      lastName: $lastName
+      password: $password
+      userRoleId: 3
+    ) {
+      id
+    }
+  }
+`;
+const QUERY_USERS = gql`
+{
+  users {
+    username
+    id
+    password
+  }
+}
+`;
+
+const authContext = createContext();
+
+const useAuth = () => useContext(authContext);
+
+function useProvideAuth() {
+  const [user, setUser] = useState(localStorage.getItem('user') || 0);
+  const { refetch } = useQuery(QUERY_USERS);
+  const [addUser] = useMutation(REGISTER);
+
+  const signIn = async (username, password) => {
+    const result = await refetch();
+    if (!result) return;
+    const match = result.data.users.find((element) => element.username === username);
```

Comment:
For one liner array functions like map, find, filter,.. you can use one letter variable name to shorten the code a bit since we don't need a meaningful name for it in this case, the condition can still be self explanatory like this:
`result.data.users.find((u) => u.username === username)`